plugins/router/freebox

```diff
@@ -0,0 +1,316 @@
+			echo 'switch.label Switch'
+			echo 'switch.cdef switch,8000,*'
+			echo 'switch.negative switch_down'
+			;;
+		.uptime)
+			echo 'graph_title Uptimes'
+			echo 'graph_category network'
+			echo 'graph_vlabel days'
+			echo 'graph_args --logarithmic'
+
+			echo 'freebox.label Freebox'
+			echo 'freebox.draw AREA'
+			echo 'freebox.cdef freebox,86400,/'
+
+			echo 'connection.label Connection'
+			echo 'connection.cdef connection,86400,/'
+			;;
+		.users)
+			echo 'graph_title Network users'
+			echo 'graph_category network'
+			echo 'graph_vlabel count'
+
+			echo 'dhcp.label DHCP leases'
+			echo 'redirect_tcp.label TCP redirections'
+			echo 'redirect_udp.label UDP redirections'
+			;;
+		*)
+			echo 'graph_title Freebox connection'
+			echo 'graph_category network'
+			echo 'graph_vlabel bits per second in (+) / out (-)'
+			echo 'graph_order main_atm_down main_atm main_wan_down main_wan'
+
+			# XXX: parameters duplicated from traffic
+			echo 'main_wan_down.label WAN down'
+			echo 'main_wan_down.graph no'
+			echo 'main_wan_down.cdef main_wan_down,8000,*'
+			echo 'main_wan.label WAN'
+			echo 'main_wan.negative main_wan_down'
+			echo 'main_wan.cdef main_wan,8000,*'
+
+			echo 'main_atm_down.label ATM down'
+			echo 'main_atm_down.graph no'
+			echo 'main_atm_down.cdef main_atm_down,1000,*'
+			echo 'main_atm.label ATM sync'
+			echo 'main_atm.negative main_atm_down'
+			echo 'main_atm.cdef main_atm,1000,*'
+			;;
+
+	esac
+	echo
+}
+
+graph_data() {
+	graph=""
+	if [ -n "$1" ]; then
+		graph=".$1"
+	fi
+
+	echo "multigraph freebox${graph}"
+	case "$graph" in
+		.adsl)
+			echo "noise.value ${NOISE_UP}"
+			echo "noise_down.value ${NOISE_DOWN}"
+
+			echo "attenuation.value ${ATTENUATION_UP}"
+			echo "attenuation_down.value ${ATTENUATION_DOWN}"
+			;;
+		.adsl_errors)
+			echo "fec.value ${FEC_UP}"
+			echo "fec_down.value ${FEC_DOWN}"
+
+			echo "crc.value ${CRC_UP}"
+			echo "crc_down.value ${CRC_DOWN}"
+
+			echo "hec.value ${HEC_UP}"
+			echo "hec_down.value ${HEC_DOWN}"
+			;;
+		.traffic)
+			echo "atm.value ${ATM_UP}"
+			echo "atm_down.value ${ATM_DOWN}"
+
+			echo "wan.value ${WAN_UP}"
+			echo "wan_down.value ${WAN_DOWN}"
+
+			echo "eth.value ${ETH_UP}"
+			echo "eth_down.value ${ETH_DOWN}"
+
+			echo "usb.value ${USB_UP}"
+			echo "usb_down.value ${USB_DOWN}"
+
+			echo "switch.value ${SWITCH_UP}"
+			echo "switch_down.value ${SWITCH_DOWN}"
+			;;
+		.uptime)
+			echo "freebox.value ${FREEBOX_UPTIME}"
+			echo "connection.value ${CONNECTION_UPTIME}"
+			;;
+		.users)
+			echo "dhcp.value ${DHCP_CLIENTS}"
+			echo "redirect_tcp.value ${REDIRECT_TCP}"
+			echo "redirect_udp.value ${REDIRECT_UDP}"
+			;;
+		*)
+			echo "main_wan.value ${WAN_UP}"
+			echo "main_wan_down.value ${WAN_DOWN}"
+
+			echo "main_atm.value ${ATM_UP}"
+			echo "main_atm_down.value ${ATM_DOWN}"
+
+
+	esac
+	echo
+}
+
+main() {
+	case $1 in
+		autoconf)
+			if curl -qso /dev/null $fbx_info_url; then
+				echo 'yes'
+			else
+				echo 'no'
```

Comment:
Please provide an explanation for `no`, e.g.: `no (failed to retrieve '$fbx_info_url')`